llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -8020,6 +8020,10 @@ Instruction *InstCombinerImpl::visitFCmpInst(FCmpInst &I) {
       if (Instruction *NV = foldOpIntoPhi(I, cast<PHINode>(LHSI)))
         return NV;
       break;
+    case Instruction::Select:
+      if (Instruction *NV = FoldOpIntoSelect(I, cast<SelectInst>(LHSI)))
```

Comment:
Should we update `constantFoldOperationIntoSelectOperand` to recognize `fcmp`?

Reply:
It should be done in a separate patch.
Related patch: https://reviews.llvm.org/D146349